checkov/secrets/plugins/ignore_secret.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from detect_secrets.util.filetype import FileType
+
+if TYPE_CHECKING:
+    from detect_secrets.util.code_snippet import CodeSnippet
+
+
+def ignore_terraform_data_block(
+    secret_line_index: int,
+    raw_context: CodeSnippet,
+) -> bool:
+    """Check for a possible data block usage"""
+
+    # search backwards to find a possible 'data' block
+    for line_index in range(secret_line_index - 1, -1, -1):
+        if raw_context.lines[line_index].lstrip().startswith('data "'):
+            # a data block is typically used to get remote information,
+            # therefore can retrieve a secret, but has not a hardcoded secret
+            return True
+
+    return False
+
+
+def ignore_secret(
```

Comment:
If we adding a mechanism for ignoring some data- maybe it can be added to the multiline parser as a function for all classes?
that's the same as skipping comments, but in this case, that's skipping specific blocks with data as a prefix.

Reply:
yeah, that was my problem, it is related to a specific file type, but I was not sure, if it is a good idea to add it to the multi parser, but I will revisit it 🤔